### DISCUSSION TITLE
Why parameters transform doesn't change the constructor private/public modifier?

### DISCUSSION THREAD

**Discussion by benyaminl** [babel/babel#16050] [Q&A]
Related to https://babeljs.io/docs/babel-plugin-transform-parameters

I just want to ask, why the babel transform the ts file, but doesn't remove private code, and still put it there, rather than put it into the function? 

I tried on babel web, it behave different and works well... : https://babeljs.io/repl#?browsers=chrome%2058&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=HYQwtgpgzgDiDGEAEBBGMkG8lIFA4EtgAXCAJwDMFkQAjeLfHPHAXyafgBsQookAogA9wMLhCaYmOeAHtgUYmQCu8YrLIAKGGQIA3EKSQATQyABcSOvACU0xsyTs2uVkA&debug=false&forceAllTransforms=true&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.23.2&externalPlugins=&assumptions=%7B%7D

and 

https://babeljs.io/repl#?browsers=chrome%2058&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=HYQwtgpgzgDiDGEAEBBGMkCgDemn6QBsIAXJEdJAXnOAHMBXQkAJwDowB7AEyYgAoA5ACFOJEAGsAKhAAeJALI8-ggDRIA2gF0AlAG5MeAvGZQoSUeOlySAYU5gYnYBGBlcBT0njOoJFgzwJJws_DAsAJYAbiAkyAAkUD4wEDpGXkgeGV4-wFCcxGyEnHT8JAAWEVBsicmpBtlIAL6GjTAMAEaEEfBIdKQAcuACaY1ZjfgA9JPevgUQRSVllVD66RMspAwswEgVVRqCoJCCWkgA_OdIAEQAEiDF13pYjS2eLevEZD6Ozq4kAC5aIxmOwAJL2X4uNwAeRgJAivmomXWng6EWA3AxdCgQPGE2OECB1wAPNdUQQmqoKfhcv4CsQWEDLJIZPJIU5oSRqY04o5mHEAKosQhAgAG8WwaBgbFyADMInQ2HyYAKIAAFWLlJqTDpiSQAWji8gNP05_11-us8jY5RIYEIYvWH08FBlZr-bn41z1VjZJGu6g9XLWTSAA&debug=false&forceAllTransforms=true&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.23.2&externalPlugins=&assumptions=%7B%7D

But on my machine, the code trans-piled is different unlike the on web : 
Source code : 
```typescript
namespace App 
{
    let app = angular.module('BotakTextModule', []);

    class BotakTextComponent {
        constructor(private $scope)
        {
            console.log(this.$scope);
        }

        public getName()
        {
            // console.log(this);
            return this['name'] ?? "Halo"; 
        }
    }

    let component: angular.IComponentOptions = {
        bindings: {
            name: "<"
        },
        controller: BotakTextComponent,
        templateUrl: `${App.config.templatePath}/botak-text-component/botakText.html`
    }

    app.component("botakText", component);
}
```

Transpiled by babel
```js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
var App;
(function (_App) {
  var app = angular.module('BotakTextModule', []);
  var BotakTextComponent = /*#__PURE__*/function () {
    function BotakTextComponent(private $scope) {
      _classCallCheck(this, BotakTextComponent);
      console.log(this.$scope);
    }
    _createClass(BotakTextComponent, [{
      key: "getName",
      value: function getName() {
        var _this$name;
        // console.log(this);
        return (_this$name = this['name']) !== null && _this$name !== void 0 ? _this$name : "Halo";
      }
    }]);
    return BotakTextComponent;
  }();
  var component = {
    bindings: {
      name: "<"
    },
    controller: BotakTextComponent,
    templateUrl: "".concat(App.config.templatePath, "/botak-text-component/botakText.html")
  };
  app.component("botakText", component);
})(App || (App = {}));
```

The babel.config.json 
```
{
    "targets": {
        "chrome": "58",
        "ie": "11"
    }
}
```

The package.json : 
```
{
  "name": "angular1",
  "version": "1.0.0",
  "description": "",
  "main": "app.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@babel/cli": "^7.23.0",
    "@babel/core": "^7.23.2",
    "@babel/plugin-syntax-jsx": "^7.22.5",
    "@babel/preset-env": "^7.23.2",
    "@babel/preset-typescript": "^7.23.2",
    "@types/angular": "^1.8.6",
    "@types/angular-route": "^1.7.4"
  }
}
```

Command used for transpiling  : 
```
npx babel --watch src --out-dir assets/js --extensions .ts --presets=@babel/preset-typescript,@babel/preset-env --source-maps inline --copy-files
```

This is angularjs legacy app and well.. it's airgap-ed, so I know it's a security issue, but for now we are delivering fixes and moving to angular 2, but it takes time. I hope someone from community can help point out what is wrong with my config

**Comment by benyaminl** [comment]
@see https://babeljs.io/docs/babel-preset-env#forcealltransforms
Ugh seems the config need to be changed to 

```json 
{
    "presets": [
        [
            "@babel/preset-env",
            {
                "targets": {
                    "chrome": "58",
                    "ie": "11"
                },
                // for uglifyjs...
                "forceAllTransforms": true
            }
        ]
    ]
}
```

**Reply by JLHwung** [reply]
The forceAllTransform option does not remove the private modifier. Instead it's the preset-typescript' work. 
In the posted configuration you have installed preset-typescript but didn't include it in your babel config, while you have enabled preset-typescript on REPL. I am surprised that Babel parser does not throw an error in this case.

**Reply by benyaminl** [reply]
@JLHwung I need to add it into the config? But I already define it on the cli using `--presets=@babel/preset-typescript,@babel/preset-env`, is it different?
